linkerd/http/prom/src/record_response/response.rs

```diff
@@ -14,26 +13,20 @@ use std::{
 };
 use tokio::{sync::oneshot, time};
 
-use super::{DurationFamily, MkDurationHistogram, MkStreamLabel};
+use super::{DurationFamily, MkDurationHistogram, MkStreamLabel, StreamLabel};
 
+/// Metrics type that tracks completed responses.
 #[derive(Debug)]
-pub struct ResponseMetrics<DurL, StatL> {
-    duration: DurationFamily<DurL>,
-    statuses: Family<StatL, Counter>,
+pub struct ResponseMetrics<L: StreamLabel> {
+    duration: DurationFamily<L::DurationLabels>,
+    statuses: Family<L::StatusLabels, Counter>,
```

Comment:
here's another core idea in this branch i'd like to highlight:

in the outbound proxy code today, we define an alias tying `DurL` and `StatL` to the same `L: StreamLabel` implementation. so, the two parameters here aren't giving us any benefit or serving any need, and we can simplify the code in this submodule (_and the equivalent `request.rs` sibling_) by making this generic across `L` rather than its underlying label types.